tests/table/json/test_json_query.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import annotations
+
+import pytest
+
+from daft.expressions import col
+from daft.table import MicroPartition
+
+
+# Chose a non-exhaustive sample of the filters provided at https://github.com/01mf02/jaq/tree/main
+@pytest.mark.parametrize(
+    "data, query, expected",
+    [
+        # Test with object
+        pytest.param(['{"col": 1}', '{"col": 2}', '{"col": 3}', None], ".col", ["1", "2", "3", None], id="object"),
+        # Test with nested objects
+        pytest.param(
+            ['{"col": {"a": 1}}', '{"col": {"a": 2}}', '{"col": {"a": 3}}', None],
+            ".col.a",
+            ["1", "2", "3", None],
+            id="nested object",
+        ),
+        # Test with array
+        pytest.param(["[1, 2, 3]", "[4, 5, 6]", "[7, 8, 9]", None], ".[1]", ["2", "5", "8", None], id="array"),
+        # Test with nested arrays
+        pytest.param(
+            ["[[1, 2, 3]]", "[[4, 5, 6]]", "[[7, 8, 9]]", None], ".[0].[1]", ["2", "5", "8", None], id="nested_array"
+        ),
+        # Test length
+        pytest.param(['"1"', "[1, 2]", '{"a": 3, "b": 4, "c": 5}'], "length", ["1", "2", "3"], id="length"),
+        # Test split
+        pytest.param(
+            ['"a,b,c"', '"d,e,f"', '"g,h,i"'],
+            'split(",")',
+            ['["a","b","c"]', '["d","e","f"]', '["g","h","i"]'],
+            id="split",
+        ),
+        # Test map
+        pytest.param(["[0, 1, 2, 3]"], "map(.*2) | [.[] | select(. < 5)]", ["[0,2,4]"], id="map"),
+        # Test iteration
+        pytest.param(["[1, 2, 3]", "[4, 5, 6]", "[7, 8, 9]"], ".[]", ["1, 2, 3", "4, 5, 6", "7, 8, 9"], id="iteration"),
```

Comment:
Should this be the intended behavior for the multiple-return case? All other executions return JSON-compatible strings, but I believe `1, 2, 3` is not JSON-compatible because it is not wrapped in a list.

What would jq return on the CLI in this case? Is it `\n`-delimited?

Reply:
yeah its `\n`-delimited

Reply:
i think we could wrap any multiple-returns in a list? like `"[1, 2, 3]"`

Reply:
Hmm I feel like my ideal would be either:

1. `\n`-delimited (to keep with the CLI behavior, also it's still valid line-delimited JSON)
2. A bit of a stretch, but if there's a way for us to know that we're getting multiple outputs by parsing the query, maybe we could do a `List[string]` return instead?

Reply:
option 1 seems the simplest

i think option 2 is not very straightforward, the jaq crate doesn't provide any methods to inspect the filter, and if we were to implement this ourselves i think it would be a lot of work.

how about 3, we just default to `List[string]` regardless of single or multiple outputs?